pkgs/development/tools/clang-tools/python.nix

```diff
@@ -0,0 +1,60 @@
+{ lib
+, stdenvNoCC
+, makeWrapper
+, xorg
+, clang-tools
+, llvmPackages
+, python3
+}:
+
+let
+  inherit (llvmPackages) clang-unwrapped;
+in
+stdenvNoCC.mkDerivation {
+  pname = "clang-tools-python";
+  inherit (clang-unwrapped) version;
+
+  dontUnpack = true;
+
+  nativeBuildInputs = [
+    makeWrapper
+    xorg.lndir
+  ];
+
+  clang_unwrapped_python = clang-unwrapped.python;
+  pythonInterpreter = python3.interpreter;
+
+  # During install, we provides the clang-tools utilities from `clang-tools` through PATH-prefixing
+  # to make the Python scripts run out-of-the-box.
+  # This changes the default behavior for them to pick up the command from the environment.
+  # For example, git-clang-format from the upstream looks for the `clang-format` command from the environment by default,
+  # and now we provide a wrapper to feed the `clang-format` command from `clang-tools` through PATH-prefixing.
+  # The binary can still be specified as `git-clang-format --binary path/to/clang-format`.
+
+  installPhase = ''
+    runHook preInstall
+
+    # Link and wrap python scripts (copy-free replacement to patchShebangs)
+    mkdir -p $out/bin
+    substituteAll ${./wrapper-python} "$out/bin/.python3-script-wrapper"
+
+    substituteInPlace "$out/bin/.python3-script-wrapper" \
+      --replace "###PATH_PREFIXING_COMMAND###" "${lib.makeBinPath [ clang-tools ]}\''${PATH:+:}\''${PATH-}"
+    chmod +x $out/bin/.python3-script-wrapper
+
+    for tool in $clang_unwrapped_python/bin/*; do
+      ln -s "$out/bin/.python3-script-wrapper" "$out/bin/$(basename "$tool")"
+    done
+
+    # Python script executables may depend on modules stored in "''${clang-unwrapped.python}/share"
+    mkdir -p $out/share
+    lndir -silent "$clang_unwrapped_python/share" "$out/share"
+
+    runHook postInstall
+  '';
+
+  meta = clang-unwrapped.meta // {
+    description = "Standalone Python scripts provided by Clang";
```

Comment:
What about homepage?

Reply:
The `homepage` attribute is inherited from `clang-unwrapped.meta`.

`clang-tools` and `clang-tools-python` are merely wrappers to the corresponding files provided by the `clang-unwrapped` package, so the homepage should be the same.

Reply:
Update: I have changed my mind after seeing `meta.name` and `meta.mainProgram` from clang-unwrapped.meta.name`.

The attributes to inherit should be explicitly specified as you suggested for the `raspa` PR (https://github.com/NixOS/nixpkgs/pull/186127#discussion_r1039016406).